crates/nu-cli/src/commands/to_xml.rs

```diff
@@ -0,0 +1,139 @@
+use crate::commands::WholeStreamCommand;
+use crate::prelude::*;
+use nu_errors::{ShellError};
+use nu_protocol::{Primitive, ReturnSuccess, Signature, SyntaxShape, UntaggedValue, Value};
+use std::io::Write;
+
+pub struct ToXML;
+
+#[derive(Deserialize)]
+pub struct ToXMLArgs {
+    pretty: Option<Value>,
+}
+
+#[async_trait]
+impl WholeStreamCommand for ToXML {
+    fn name(&self) -> &str {
+        "to xml"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("to xml").named(
+            "pretty",
+            SyntaxShape::Int,
+            "Formats the XML text with the provided indentation setting",
+            Some('p'),
+        )
+    }
+
+    fn usage(&self) -> &str {
+        "Convert table into .xml text"
+    }
+
+    async fn run(
+        &self,
+        args: CommandArgs,
+        registry: &CommandRegistry,
+    ) -> Result<OutputStream, ShellError> {
+        to_xml(args, registry).await
+    }
+}
+
+pub fn write_xml_events<W: Write>(value: &Value, writer: &mut xml::writer::EventWriter<W>) -> Result<(), ShellError> {
+    Ok(match &value.value {
+        UntaggedValue::Primitive(Primitive::Boolean(b)) => {
+            writer.write(b.to_string().as_str()).unwrap();
+        },
+        UntaggedValue::Row(o) => {
+            for (k, v) in o.entries.iter() {
+                writer.write(xml::writer::XmlEvent::start_element(k.as_str())).unwrap();
+                write_xml_events(v, writer).unwrap();
+                writer.write(xml::writer::XmlEvent::end_element()).unwrap();
+            }
+        },
+        UntaggedValue::Table(l) => {
+            writer.write(xml::writer::XmlEvent::start_element("table")).unwrap();
```

Comment:
That's mostly a test to see how structured data can be serialised with `xml-rs`. Don't intend to leave it like that.

Reply:
Just make sure to convert your `unwrap`s to `expect`s, `clippy` will fail your builds otherwise in the CI